### PR TITLE
Fix: Default app size now shows normally and other Adaptive fixes plus responsive fix on default

### PR DESCRIPTION
**Pull request by JoseMoreville**

### CODE REVIEW

**Comment by JoseMoreville** [comment]
Some cleaning at the code is necessary, but tests are welcome

**Comment by Depal1** [comment]
Tests of https://github.com/PlayCover/PlayTools/pull/77 (https://github.com/PlayCover/PlayTools/pull/77/commits/2a8b8249d4da01541ed13594cad3b2afd0eaba22) with https://github.com/PlayCover/PlayCover/pull/781 (  https://github.com/PlayCover/PlayCover/pull/781/commits/b0e62b1aaa1be7e638a3653b67133156f5b1b091) on macOS 13.1.

### Angry Birds Space *com.rovio.AngryBirdsSpace*
- App Default
AB Space App Default
- 1080p 16:9 [Fix windows issues *off*]
AB Space 1080p off
- 1080p 16:9 [Fix windows issues *on*]
AB Space 1080p on

### Dadish *com.thomasyoung.gooddadish*
- App Default
Dadish Default
- 1080p 16:9 [Fix windows issues *off*]
Dadish 1080p off

### Genshin Impact *com.miHoYo.GenshinImpact*
- App Default
Genshin Default

**Comment by JoseMoreville** [comment]
@Depal1 is it possible to test this?

**Review by Depal1 — CHANGES_REQUESTED** [review]
All tests where performed on macOS 13.1 with the commits https://github.com/PlayCover/PlayTools/pull/77/commits/fd6dfbd18ac6893950b009a0aaf16ae9cfd700bf and https://github.com/PlayCover/PlayCover/pull/781/commits/e74e6978fa06e725acfd8691b60a1ef485b22e91 of PlayTools and PlayCover respectively.

App default seems to work as intended on macOS 13.1, both in resizable windows and fixed frames.

The regression regarding improper view window position in non-app-default settings, introduced in changes after the 2.0.7 PlayTools tag for macOS 13.1 and previous versions, is still present on Angry Birds Space. It is unbeknownst to me if it is present on 13.2 too.

A similar regression is present on Plague Inc., which can't be displayed correctly with either `Fix window issues` *on* or *off*, showing in both cases as in the following picture (1080p 16:9).

Plague 1080p 16:9

Tests were also made changing the aspect ratio, in order to visually confirm that window settings were indeed being changed on a general basis. Hence, `Fix window issues` might not be working correctly. Further on this hypothesis, Angry Birds Space is showing in the same way when either `Fix window issues` is *on* or *off*, in contraposition to the behaviour in previous tests.

Rather than checking for macOS 13.2 from PlayCover and giving that information to PlayTools, those checks could be done from PlayTools itself using the corresponding check availability attributes both in Swift and Objective-C, using the equivalent iOS 16.3 check for macOS 13.2 since PlayTools is iOS and then converted via Mac Catalyst. https://developer.apple.com/documentation/swift/marking-api-availability-in-objective-c

It is encouraged to test such an approach first. Otherwise, the purpose of splitting PlayCover from PlayTools is partly defeated by making PlayTools strictly dependent from PlayCover, and worthless as a standalone framework unnecessarily. The former instance has the potential to reduce code too.

**Comment by JoseMoreville** [comment]
The toggle Fix window issues is not meant to work in 13.1 just 13.2 and up. sorry that I forgot to add that check for that in Frontend

File: PlayTools/PlayScreen.swift [hunk 1]
```diff
@@ -20,7 +21,7 @@ extension CGSize {
     }
 
     func toAspectRatio() -> CGSize {
-        if #available(iOS 16.3, *) {
+        if macOSVersion >= 13.19000 {
```
[the remark this answers is not in the document]
Comment (JoseMoreville) [reply PlayTools/PlayScreen.swift:24]: This will be added in next PR.

**Comment by JoseMoreville** [comment]
@Depal1 sorry, the toggle for fix screen issue is not meant to work for 13.1 just 13.2 and up.
I added that toggle for default screen res too as a way to fix apps that are responsive by default like Twitch, LIFX, Paperback or Aidoku.

File: PlayTools/PlaySettings.swift [hunk 2]
```diff
@@ -33,14 +33,18 @@ let settings = PlaySettings.shared
...
-    lazy var windowSizeWidth = CGFloat(settingsData.windowWidth)
+    @objc lazy var windowSizeWidth = CGFloat(settingsData.windowWidth)
+
+    @objc lazy var inverseScreenValues = settingsData.inverseScreenValues
 
     @objc lazy var adaptiveDisplay = settingsData.resolution == 0 ? false : true
 
     @objc lazy var deviceModel = settingsData.iosDeviceModel as NSString
 
+    @objc lazy var macOSVersion = Double(settingsData.macOSVersion)
```
Comment (Depal1) [inline PlayTools/PlaySettings.swift:46]:
```suggestion

```

File: PlayTools/PlaySettings.swift [hunk 3]
```diff
@@ -82,4 +86,6 @@ struct AppSettingsData: Codable {
     var version = "2.0.0"
     var playChain = false
     var playChainDebugging = false
+    var inverseScreenValues = false
+    var macOSVersion: Double = 13.2
```
Comment (Depal1) [inline PlayTools/PlaySettings.swift:90]:
```suggestion

```

**Review by Depal1 — COMMENTED** [review]
It is working correctly both on macOS 13.1 and 13.2. Since the iOS versions checks are working, the remains of the former check are no longer needed.

**Review by Depal1 — APPROVED** [review]

**Pull request merged**